x-pack/plugins/search_connectors/server/services/index.ts

```diff
@@ -0,0 +1,230 @@
+            `Skipping connector ${connector.id}: unsupported service type ${connector.service_type}`
+          );
+          continue;
+        }
+
+        nativeConnectors.push({
+          id: connector.id,
+          name: connector.name,
+          service_type: connector.service_type,
+        });
+      }
+    }
+    return nativeConnectors;
+  };
+
+  public getConnectorPackagePolicies = async (): Promise<PackagePolicyMetadata[]> => {
+    const kuery = `${PACKAGE_POLICY_SAVED_OBJECT_TYPE}.package.name:${pkgName}`;
+    this.logger.debug(`Fetching policies with kuery: "${kuery}"`);
+    const policiesIterator = await this.packagePolicyService.fetchAllItems(this.soClient, {
+      perPage: 50,
+      kuery,
+    });
+    const policiesMetadata: PackagePolicyMetadata[] = [];
+    for await (const policyPage of policiesIterator) {
+      for (const policy of policyPage) {
+        for (const input of policy.inputs) {
+          if (input.type === connectorsInputName) {
+            if (input.compiled_input != null) {
+              if (input.compiled_input.service_type == null) {
+                this.logger.debug(`Policy ${policy.id} is missing service_type, skipping`);
+                continue;
+              }
+
+              if (input.compiled_input.connector_id == null) {
+                this.logger.debug(`Policy ${policy.id} is missing connector_id, skipping`);
+                continue;
+              }
+
+              if (input.compiled_input.connector_name == null) {
+                this.logger.debug(`Policy ${policy.id} is missing connector_name`);
+                // No need to skip, that's fine
+              }
+
+              if (policy.supports_agentless !== true) {
+                this.logger.debug(`Policy ${policy.id} does not support agentless, skipping`);
+                continue;
+              }
+
+              policiesMetadata.push({
+                package_policy_id: policy.id,
+                agent_policy_ids: policy.policy_ids,
+                connector_metadata: {
+                  id: input.compiled_input.connector_id,
+                  name: input.compiled_input.connector_name || '',
+                  service_type: input.compiled_input.service_type,
+                },
+              });
+            }
+          }
+        }
+      }
+    }
+
+    return policiesMetadata;
+  };
+
+  public deployConnector = async (connector: ConnectorMetadata): Promise<PackagePolicy> => {
+    this.logger.info(
+      `Connector ${connector.id} has no integration policy associated with it, creating`
+    );
+
+    if (connector.id == null || connector.id.trim().length === 0) {
+      throw new Error(`Connector id is null or empty`);
+    }
+
+    if (connector.service_type == null || connector.service_type.trim().length === 0) {
+      throw new Error(`Connector ${connector.id} service_type is null or empty`);
+    }
+
+    if (NATIVE_CONNECTOR_DEFINITIONS[connector.service_type] == null) {
```

Comment:
Using our regular `NATIVE_CONNECTOR_DEFINITIONS` as a source of truth for connectors that we support. I could theoretically instead list integrations that are branched off `connectors-py` instead, is it possible/better?